chat/QuickJump.vue

```diff
@@ -0,0 +1,459 @@
+          e.preventDefault();
+          if (
+            this.filteredResults.length === 0 &&
+            this.searchQuery.length > 0
+          ) {
+            // Only the "open new conversation" option is available
+            this.selectedIndex = -1;
+          } else if (this.selectedIndex === -1) {
+            // From "open new conversation" to last result
+            this.selectedIndex = this.filteredResults.length - 1;
+          } else if (this.selectedIndex > 0) {
+            this.selectedIndex--;
+          } else {
+            // From first result, go to "open new conversation" if available, otherwise wrap to last
+            if (this.searchQuery.length > 0) {
+              this.selectedIndex = -1;
+            } else {
+              this.selectedIndex = this.filteredResults.length - 1;
+            }
+          }
+          break;
+
+        case Keys.ArrowDown:
+          e.preventDefault();
+          if (
+            this.filteredResults.length === 0 &&
+            this.searchQuery.length > 0
+          ) {
+            // Only the "open new conversation" option is available
+            this.selectedIndex = -1;
+          } else if (this.selectedIndex === -1) {
+            // From "open new conversation" to first result
+            this.selectedIndex = 0;
+          } else if (this.selectedIndex < this.filteredResults.length - 1) {
+            this.selectedIndex++;
+          } else {
+            // From last result, go to "open new conversation" if available, otherwise wrap to first
+            if (this.searchQuery.length > 0) {
+              this.selectedIndex = -1;
+            } else {
+              this.selectedIndex = 0;
+            }
+          }
+          break;
+
+        case Keys.Enter:
+          e.preventDefault();
+          if (
+            this.selectedIndex === -1 &&
+            this.filteredResults.length === 0 &&
+            this.searchQuery.length > 0
+          ) {
+            // Open new conversation
+            this.openNewConversation();
+          } else if (this.filteredResults.length > 0) {
+            this.selectResult(this.filteredResults[this.selectedIndex]);
+          }
+          break;
+      }
+    }
+
+    selectResult(result: SearchResult): void {
+      if (result.conversation) {
+        result.conversation.show();
+      }
+      this.hide();
+    }
+
+    openNewConversation(): void {
+      if (this.searchQuery.trim().length === 0) return;
+
+      // Create a character object for the username
+      const character = core.characters.get(this.searchQuery.trim());
+
+      // Create and show the conversation
+      const conversation = core.conversations.getPrivate(character);
+      conversation.show();
+
+      this.hide();
+    }
+
+    @Watch('visible')
+    onVisibilityChange(newValue: boolean): void {
+      if (newValue) {
+        // Update results when showing
+        this.updateResults();
+      }
+    }
+  }
+</script>
+
+<style lang="scss" scoped>
```

Comment:
I _love_ the look for this, but it really doesn't fit in with the rest of the Bootstrap-styled components. I don't think we should change it now, but if other people are embracing this look too we'll have to consider generalising some of the CSS here.